src/pgduckdb_ddl.cpp

```diff
@@ -416,6 +439,9 @@ DECLARE_PG_FUNCTION(duckdb_create_table_trigger) {
 		PG_RETURN_NULL();
 	}
 
+	/* Reset since we don't need it anymore */
+	pgduckdb::top_level_duckdb_ddl_type = pgduckdb::DDLType::NONE;
```

Comment:
~How is it reset if an exception occurs before?~ Actually expanding more the code it seems that it is reset because we're starting the function? Is the comment above correct?

Reply:
Changed the logic now and updated the comment. On failure (and also on success to be safe) this value is also reset in `DuckdbXactCallback_Cpp` in `src/pgduckdb_xact.cpp`.